src/workers/postUpdated.ts

```diff
@@ -165,12 +175,45 @@ const createPost = async ({
     ? PostOrigin.CommunityPicks
     : data.origin ?? PostOrigin.Crawler;
 
-  const post = await entityManager.getRepository(ArticlePost).create(data);
+  const post = await entityManager
+    .getRepository(contentTypeFromPostType[data.type])
+    .create(data);
   await entityManager.save(post);
 
   await addKeywords(entityManager, mergedKeywords, data.id);
   await addQuestions(entityManager, questions, data.id);
 
+  let relatedCollections: CollectionPost[] = [];
+
+  if (post.type === PostType.Collection) {
+    await addRelatedPosts({
+      entityManager,
+      postId: post.id,
+      entries: collectionRelatedPosts,
+      relationType: PostRelationType.Collection,
+    });
+  } else if (collectionRelatedPosts) {
+    relatedCollections = await relatePosts({
+      entityManager,
+      postId: post.id,
+      entries: collectionYggdrasilIds,
+      relationType: PostRelationType.Collection,
+    });
+  }
+
+  const isCollectionUpdated =
+    post.type === PostType.Collection || relatedCollections.length > 0;
+
+  if (isCollectionUpdated) {
+    const collectionPosts = [post, ...relatedCollections];
+
+    await Promise.allSettled(
+      collectionPosts.map((collectionPost) => {
+        return notifyPostCollectionUpdated(logger, collectionPost);
+      }),
+    );
+  }
```

Comment:
Why don't we use CDC for this purpose? Wouldn't it be safer?

Reply:
CDC on post relation?

Reply:
Yes

Reply:
Another approach is whenever we update the post relation to update also the `metadataUpdatedAt` field and to catch that on a CDC

Reply:
If we did it on post relation it would be multiple CDC messages in the same time. That would not be efficient. 

For CDC on post, sometimes the post is a collection and we add relations to it, and sometimes post is part of multiple collections and then we add it as a relation. So we would be basically updating `metadataUpdatedAt` for multiple posts at the same time again. That field until this point is handled by the platform side which could then lead to missed updates because our `metadataUpdatedAt` would be newer then platform one. 

I hope it makes sense but I think the separate subscription for now is better.

Reply:
My concern is not the separate subscription but the fact we introduce a distributed transaction without being aware. Meaning we have a function that assumes a database change + pubsub notification are atomic which is not the case. Ideally we want to do only database changes which will trigger CDC which will trigger whatever business logic we need.
Another option is to always update the `metadataUpdatedAt` of the collection on both cases.
Anyway, happy for any ideas to solve the issue I raised

Reply:
What would be your concrete suggestion in that case? Perhaps it's worth discussing it on Slack, so others can contribute or in the DR.

Reply:
@idoshamun I did a change in 9b287bf3404eb7b0ee584e85866e49b1220ecd11 to use post_relation cdc.